### PR TITLE
Add state resource receipt proof contract

### LINKED ISSUE #2069: [PM] Add state resource receipt proof contract
task_uid: task_f7ed2407364b466b930d8f1633f4f111

GitHub-backed oasis7 PM task.

Task metadata:
- owner_role: `tpm`
- module: `p2p`
- status: `pr_watch`
- priority: `P1`
- worktree_hint: `/Users/scc/ccwork/worktrees/oasis7-p2p-state-resource-receipt-proof-contract`
- pr_url: `https://github.com/eng-cc/oasis7/pull/2079`
- pr_number: `2079`

Source refs:
- `doc/p2p/blockchain/p2p-current-mainstream-public-chain-gap-benchmark-2026-07-03.md`
- `doc/testing/templates/chain-proof-evidence-v1-template.md`

Acceptance:
- Define a minimal state/resource/receipt proof contract that binds concrete resource/query/receipt evidence to an already committed world head or state root.
- Add verifier coverage or validation tests for positive and fail-closed negative cases without claiming full light-client, mainnet, or multi-client equivalence.
- Update benchmark/project/runbook docs with the new bounded proof status and remaining gaps.

### PR DESCRIPTION
**Pull request by eng-cc**
Task: task_f7ed2407364b466b930d8f1633f4f111

Closes #2069

Generated by ./scripts/prepare-task-pr.sh.

### CODE REVIEW

File: crates/oasis7_proto/src/distributed_state_receipt.rs [hunk 1]
```diff
@@ -0,0 +1,797 @@
...
+        for (index, node) in self.proof_path.iter().enumerate() {
+            require_non_empty(
+                format!("proof_path[{index}].sibling_hash").as_str(),
+                &node.sibling_hash,
+            )?;
+            if !seen_sibling_hashes.insert(node.sibling_hash.as_str()) {
+                return Err(format!(
+                    "proof_path[{index}].sibling_hash must not duplicate an earlier node"
+                ));
+            }
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply crates/oasis7_proto/src/distributed_state_receipt.rs]: Fixed in 859dbab35: removed the sibling-hash uniqueness rejection while retaining non-empty ordered path validation and root recomputation. Added `world_state_receipt_proof_v1_accepts_repeated_sibling_hashes`. Verified with `env -u RUSTC_WRAPPER cargo test -p oasis7_proto`.

File: scripts/network-tier-public-testnet-readiness.sh [hunk 2]
```diff
@@ -592,6 +592,272 @@ def validate_chain_proof_evidence_pass_evidence(raw: str, evidence: pathlib.Path
...
+    if not isinstance(proof_targets, dict):
+        blockers.append(f"{lane} proof_targets object missing: {raw}")
+    else:
+        state_or_query = proof_targets.get("state_or_query")
+        if proof_kind in {"resource_state", "query_result"} and not isinstance(state_or_query, dict):
+            blockers.append(f"{lane} proof_targets.state_or_query object missing: {raw}")
+        elif isinstance(state_or_query, dict):
+            for key in ("proof_kind", "namespace", "root_hash", "leaf_hash", "proof_status"):
+                if not str(state_or_query.get(key) or "").strip():
+                    blockers.append(f"{lane} proof_targets.state_or_query.{key} missing: {raw}")
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply scripts/network-tier-public-testnet-readiness.sh:768]: Fixed in 859dbab35: readiness validation now requires `external_verifier.proof_status` to be supported and equal to `proof_targets.state_or_query.proof_status` for state/query proofs. Added a fail-closed mismatch case in `scripts/network-tier-chain-proof-evidence-lane.test.sh`; verified with that script.

File: scripts/network-tier-public-testnet-readiness.sh [hunk 3]
```diff
@@ -592,6 +592,272 @@ def validate_chain_proof_evidence_pass_evidence(raw: str, evidence: pathlib.Path
...
+                        blockers.append(f"{lane} proof_targets.state_or_query.namespace must match external_verifier.subject.namespace: {raw}")
+                    if proof_kind == "resource_state" and str(state_or_query.get("resource_id") or "").strip() != str(verifier_subject.get("resource_id") or "").strip():
+                        blockers.append(f"{lane} proof_targets.state_or_query.resource_id must match external_verifier.subject.resource_id: {raw}")
+                    if proof_kind == "query_result":
+                        for key in ("query_id", "query_hash"):
+                            if str(state_or_query.get(key) or "").strip() != str(verifier_subject.get(key) or "").strip():
+                                blockers.append(f"{lane} proof_targets.state_or_query.{key} must match external_verifier.subject.{key}: {raw}")
+        resource = proof_targets.get("resource")
+        if proof_kind == "resource_state" and not isinstance(resource, dict):
+            blockers.append(f"{lane} proof_targets.resource object missing: {raw}")
```
[the remark this answers is not in the document]
Comment (eng-cc) [reply scripts/network-tier-public-testnet-readiness.sh:802]: Fixed in 859dbab35: resource content/commit metadata is now required only for included `resource_state` proofs; absent resource proofs can omit `proof_targets.resource`. Added a valid absence fixture and readiness assertion; verified with `./scripts/network-tier-chain-proof-evidence-lane.test.sh`.
